views/js/runner/plugins/tools/zoom.js

```diff
@@ -0,0 +1,247 @@
+
+            function zoomAction(dir) {
+
+                var inc = increment * dir;
+                var el, sx, sy, before, after, margin;
+
+                if (self.$zoomTarget) {
+                    el = self.$zoomTarget[0];
+                    
+                    before = el.getBoundingClientRect();
+
+                    sx = self.$container.scrollLeft();
+                    sy = self.$container.scrollTop();
+
+                    self.zoom = Math.max(threshold.lower, Math.min(threshold.upper, self.zoom + inc));
+
+                    if (self.zoom === standard) {
+                        resetZoom(self.$zoomTarget);
+                    } else {
+                        setZoomLevel(self.$zoomTarget, self.zoom);
+                    }
+
+                    // force a browser repaint to fix a scrollbar issue with WebKit
+                    forceRepaint(self.$zoomTarget);
+
+                    after = el.getBoundingClientRect();
+
+                    sx = Math.max(0, sx + (after.width - before.width) / 2);
+                    sy = Math.max(0, sy + (after.height - before.height) / 2);
+
+                    self.$container.scrollLeft(sx).scrollTop(sy);
+
+                    // center the item while zooming out
+                    margin = Math.max(0, self.$zoomTarget.outerWidth() - after.width) / 2;
+                    self.$zoomTarget.css({
+                        'margin-left' : margin + 'px',
+                        'margin-right' : -margin + 'px'
+                    });
+                }
+            }
+            
+            /**
+             * Is calculator activated ? if not, then we hide the plugin
+             */
+            function togglePlugin(){
+                //var context = testRunner.getTestContext();
+                //to be activated always
+                /*if(context.options.zoom) {
+                    self.show();
+                }else{
+                    self.hide();
+                }*/
+                self.show();
+            }
+
+            //build element (detached)
+            this.$buttonZoomOut = $(buttonTpl({
+                control : 'zoomOut',
+                title : __('Zoom out'),
+                icon : 'remove',
+                text : __('Zoom out')
+            }));
+
+            this.$buttonZoomIn = $(buttonTpl({
+                control : 'zoomIn',
+                title : __('Zoom in'),
+                icon : 'add',
+                text : __('Zoom in')
+            }));
+
+            this.zoom = standard;
+            
+            //attach behavior
+            this.$buttonZoomOut.on('click', function (e){
+
+                e.preventDefault();
+                zoomAction(-1);
+            });
+
+            //attach behavior
+            this.$buttonZoomIn.on('click', function (e){
+                e.preventDefault();
+                zoomAction(1);
+            });
+            
+            //start disabled
+            togglePlugin();
```

Comment:
Please just use this.show() and remove the togglePlugin function as the plugin will be registered from client extension